resources/design.doc2.md

```diff
@@ -0,0 +1,181 @@
+  * If the project is not found, will throw a ProjectNotFoundException 
+  * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+
+## 6.2. _Update Task Information Endpoint_
+* Accepts `PUT` requests to `/tasks/:id`
+* Accepts a task ID and updated task information, Returns the updated the task with the new information.
+    * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+
+## 6.2. _Delete Task From Project Endpoint_
+* Accepts `DELETE` requests to `/tasks/:id`
+* Accepts a task ID, deletes the task with that ID from project.
+    * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+
+## 6.2. _View All Projects Endpoint_
+* Accepts `GET` requests to `/projects`
+* Accepts a project ID, returns a list of projects created by that Manager.
+    * If the given project ID doesn't exist, will throw an ``ProjectNotFoundException``
+    * If the given Manager has not created any projects, an empty list will be returned
+
+## 6.2. _View All Assigned Tasks Endpoint_
+* Accepts `GET` requests to ` /tasks/assigned`
+* Retrieves a list of all tasks assigned to given user ID.
+* Accepts a user ID, returns a list of all tasks assigned to that user.
+    * If the given user ID doesn't exist, will throw an ``UserNotFoundException``
+    * If the given user does not have any tasks, an empty list is returned.
+
+## 6.2. _View Assigned Tasks By Due Date Endpoint_
+* Accepts `GET` requests to ` /tasks/assigned/prioritized`
+* Retrieves a list of all tasks assigned to given user ID.
+  * Returns the song list in default playlist order
+    * If the optional order parameter is provided, this API will return the task list in ``order``, reverse order, or by Due Date, based on the value of ``order``
+      * DEFAULT - same as default behavior, returns tasks in ordered assigned by Manager
+      * REVERSED - returns project tasks in reversed order 
+      * DUEDATE - returns project tasks in order of due date.
+  * If the given project ID is found, but contains no tasks, the tasks list will be empty
+  * If the given project ID doesn't exist, will throw an ``ProjectNotFoundException``
+
+## 6.2. _Update Task Progress Endpoint_
+* Accepts `PUT` requests to `/tasks/:id/progress`
+* Accepts a task ID and a new progress status (Not Started, In Progress, Complete), updates the ``taskStatus`` of the task with that ID.
+    * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+
+## 6.2. _Track Time Spent on Task Endpoint_
+* Accepts `PUT` requests to `/tasks/:id/time`
+* Accepts a task ID and a new time spent on task, updates the ``taskTimer`` of the task with that ID.
+    * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+
+
+# 7. Tables
+### 7.1. `Projects`
+```
+projectId // Partition Key, String
+projectId // String
+projectName // String
+createdBy // String
+tasks // List<String>
```

Comment:
I added a question above about how tasks relate to projects. We should revisit this schema after you clarify that.